logilica_weekly_report/update_gdoc.py

```diff
@@ -44,3 +67,149 @@ def add_teams(
                 width="80%",
                 height="80%",
             )
+
+
+def upload_doc(doc: str, creds: Credentials, config: dict[str, any]) -> str:
+    """Upload the provided in-memory HTML document to Google Drive."""
+
+    filename = (
+        config.get("google", {})
+        .get("filename", DEFAULT_GDRIVE_FILE_TEMPLATE)
+        .format(datetime.now())
+    )
+    file_metadata = {
+        "name": filename,
+        "mimeType": "application/vnd.google-apps.document",
+    }
+    try:
+        service = build("drive", "v3", credentials=creds)
+        media = MediaIoBaseUpload(
+            BytesIO(doc.encode()), mimetype="text/html", resumable=True
+        )
+        request = service.files().create(
+            body=file_metadata,
+            fields="id",
+            media_body=media,
+            includeLabels=None,
+            includePermissionsForView=None,
+        )
+
+        response = None
+        while response is None:
+            status, response = request.next_chunk()
+            if status:
+                print(f"Uploaded {int(status.progress() * 100)}%.")
+        logging.info(
+            'File "%s" with ID "%s" has been uploaded.', filename, response.get("id")
+        )
+
+    except HttpError as error:
+        logging.error("An error occurred uploading %s: %s", filename, error)
+        raise
```

Comment:
The code is not handling this exception raised. That's going to stop the process. Is that expected?

Reply:
Yes, that's expected.  The document is uploaded in chunks (although, the chunks are pretty big, so that it's typically only a single iteration), and the code will retry/restart the upload in the event of an error on one of the chunk requests; so, if we reach the `except` statement at line 106, it means that the upload as a whole has failed, and there is no point in continuing.

The code here will log an error and then (re-)raise the exception which will ultimately be caught at the main function which will print an error message and exit with an error status (so there won't be an ugly traceback).